src/data_type/document.details.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const VALIDATION_PATTERN = /^([A-Za-z_]*) ([A-Za-z]{3}) ([A-Za-z0-9]{1}).*$/;
+const TYPE_INDEX = 0;
+const COUNTRY_INDEX = 1;
+const NUMBER_INDEX = 2;
+const EXPIRATION_INDEX = 3;
+const AUTHORITY_INDEX = 4;
+
+module.exports.DocumentDetails = class DocumentDetails {
+  constructor(value) {
+    this.validateData(value);
+    this.parseFromValue(value);
+  }
+
+  validateData(value) {
+    const regex = new RegExp(VALIDATION_PATTERN);
+    if (!regex.test(value)) {
+      throw new Error('Invalid value for DocumentDetails');
+    }
+  }
+
+  parseFromValue(value) {
+    this.validateData(value);
+    const parsedValues = value.split(' ');
+
+    this.type = parsedValues[TYPE_INDEX];
+    this.issuingCountry = parsedValues[COUNTRY_INDEX];
+    this.documentNumber = parsedValues[NUMBER_INDEX];
+
+    if (parsedValues.length > EXPIRATION_INDEX) {
+      const dateValue = parsedValues[EXPIRATION_INDEX];
+      const dateObj = new Date(dateValue);
```

Comment:
What happens if _dateValue_ is not a correctly formatted date value?  Won't this fail?

Reply:
Nope it will return null, hence there is isNaN check on line 35.

Reply:
Fair enough